osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -122,11 +127,7 @@ protected override void Deselected()
 
             MovementContainer.MoveToX(0, 500, Easing.OutExpo);
 
-            if (beatmapContainer != null)
```

Comment:
This null check going away worries me. `updateBeatmapYPositions()` assumes that `beatmapContainer` cannot be null. Are you sure it is safe to drop this guard?

Reply:
The only place the container is set to null is where `beatmapLoadTask` is also set to null. We can add back the null check if you'd prefer.

Reply:
I imagine a pattern like this might trigger a NRE:

```
DrawableCarouselItem.Item.set is called
DrawableCarouselBeatmapSet.UpdateItem is called by the setter (nulling out container)
DrawableCarouselItem.State changes to Deselected
DrawableCarouselBeatmapSet.Deselected is called by the state change
DrawableCarouselBeatmapSet.updateBeatmapYPositions is called, and throws on null container
```

but I'm not sure how realistic it is. I'd probably bring back the check but do it in `updateBeatmapYPositions` for safety. In the above case the positions will get updated when the beatmaps are done loading anyway.